domain/resource.go

```diff
@@ -42,6 +42,7 @@ type ListResourcesFilter struct {
 	Size          uint32            `mapstructure:"size" validate:"omitempty"`
 	Offset        uint32            `mapstructure:"offset" validate:"omitempty"`
 	OrderBy       []string          `mapstructure:"order_by" validate:"omitempty"`
+	Q             string            `mapstructure:"q" validate:"omitempty"`
 }
```

Comment:
As we are doing like over Name/urn. Isn't it better to rename it to `matches` instead of `q`. WDYT?

Reply:
q stands for query, and we already use this in list grants, approvals, and appeals. So aside from for the consistency, it is also more common to use q or query for search parameter (cmiiw). and yes it queries on resource name OR urn